internal/bufferedread/buffered_reader_test.go

```diff
@@ -166,3 +166,78 @@ func (t *BufferedReaderTest) TestCheckInvariantsNoPanic() {
 
 	assert.NotPanics(t.T(), func() { reader.CheckInvariants() })
 }
+
```

Comment:
I don't see test for scheduleBlockWithIndex.

This is important to test for different blockIndex and value of urgent.